src/checkpoint/checkpoint_api.h

```diff
@@ -94,16 +95,32 @@ SerializedReturnType serialize(T& target, BufferCallbackType fn = nullptr);
  * deallocated with \c delete
  *
  * \param[in] buf the buffer containing the bytes to reify \c T
- * \param[in] len the number of bytes in \c buf
- * \param[in] user_buf (optional) buffer containing bytes allocated with
- * sufficient size for \c T. If this buffer is passed, the caller is responsible
+ * \param[in] object_buf (optional) buffer containing bytes allocated with
+ * sufficient size for \c T. If this buffer != null, the caller is responsible
```

Comment:
I think it should now be asserted that for this function, `object_buf != nullptr`, and this should never allocate.

The other overload can trivially be implemented in terms of this one.